src/bitwarden_workflow_linter/rules/check_pr_target.py

```diff
@@ -27,16 +27,12 @@ def __init__(self, settings: Optional[Settings] = None, lint_level: Optional[Lin
         self.compatibility = [Workflow]
         self.settings = settings
 
-    def targets_main_branch(self, obj:Workflow) -> bool:
-        if obj.on["pull_request_target"].get("branches"):
-            branches_list = obj.on["pull_request_target"].get("branches")
-            if isinstance(branches_list, str):
-                branches_list = [branches_list]
-            if any(branch != 'main' for branch in branches_list):
-                return False
-        else:
-            return False
-        return True
+    def targets_main_branch(self, obj: Workflow) -> bool:
+        default_branch = self.settings.default_branch if self.settings else "main"
+        branches = obj.on["pull_request_target"].get("branches", [])
+        if isinstance(branches, str):
+            branches = [branches]
+        return branches == [default_branch]
```

Comment:
Rather than comparing a list of strs against a list of strs, it's safer to break this out into 
1. Verify the list has 1 element and if not then return false
2. Verify the lists's 1 element is equivalent to the default_branch value
```suggestion
        return len(branches) == 1 and branches[0] == default_branch
```